libsolidity/ast/AST.cpp

```diff
@@ -498,6 +498,18 @@ bool Declaration::isEventParameter() const
 	return dynamic_cast<EventDefinition const*>(scope());
 }
 
+bool Declaration::isVisibleAsUnqualifiedName() const
+{
+	if (!scope())
+		return true;
+	if (isStructMember() || isEnumValue() || isEventParameter())
```

Comment:
After #10987 this should include custom error parameters as well.

We also need a test case covering events and errors.

Reply:
Done. I have also added more tests.